### PR TITLE
The new "Discover what's new in the PIM per version" page 

### PR DESCRIPTION
**Pull request by NolwennP**
The idea of this page is to display what's new in the PIM depending on the PIM version and edition, and the PXM domains.
You can select a set of filters, and the page will display all the new key features that we delivered in the versions/edition you selected for the PXM domains you chose.

Left to do:
- [x] Review English
- [x] Add the "Read more" links to versions before the 4.0
- [x] Link to the page from the homepage
- [x] Fix bad ordering inside each version

### CODE REVIEW

File: tasks/build-versions-comparison.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+const revReplace = require('gulp-rev-replace');
+const _ = require('lodash');
+
+const majorVersion = 'serenity';
+const features = [
+    'Asset',
+    'Reference entities',
+    'Rules engine'
+];
+const domainColors = {
```
Comment (ahocquard) [inline tasks/build-versions-comparison.js:18]: I would directly put the correct format in the const for clarity, as you use it just once and transform it in another format:

```
const coloredDomains = {
   { 'color' : 'info', 'domain': 'Productivity'},
   ....
};
```

File: tasks/build-versions-comparison.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+const gulp = require('gulp');
+const hbs = require('handlebars');
+const gulpHandlebars = require('gulp-handlebars-html')(hbs);
+const fs = require('fs');
+const rename = require('gulp-rename');
+const revReplace = require('gulp-rev-replace');
+const _ = require('lodash');
+
+const majorVersion = 'serenity';
+const features = [
```
Comment (ahocquard) [inline tasks/build-versions-comparison.js:13]: same

File: tasks/build-versions-comparison.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    const versions = {};
+    _.each(orderedVersions, function(version){
+        versions[version.name.split('.').join('').toLowerCase()] = version;
+    });
+    const updates = _.reduce(_.reverse(orderedVersions), function(concatUpdates, version){
+        return concatUpdates.concat(JSON.parse(fs.readFileSync(fileDirectorySource + '/' + version.name + '-news.json')));
+    }, []);
+
+    const coloredDomains = {};
+    _.each(domainColors,function(color,domain){
```
Comment (ahocquard) [inline tasks/build-versions-comparison.js:58]: I would remove it and use the correct format directly in the const

File: tasks/build-versions-comparison.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            features: coloredFeatures,
+            defaultDomainsAndFeatures: Object.keys(domainsAndFeatures),
+            versions: versions,
+            defaultVersions: Object.keys(versions),
+            majorVersion: majorVersion
+        }, {
+            partialsDirectory: ['./src/partials']
+        }))
+        .pipe(rename('versions-comparison.html'))
+        .pipe(revReplace({manifest: gulp.src("./tmp/rev/rev-manifest.json")}))
```
Comment (ahocquard) [inline tasks/build-versions-comparison.js:92]: it renames the css to invalidate it if we change it.
But or the new JS it introduced, it does not change the name of the script. It means that if you fix the JS in a future PR (a bug for example), the JS will be in cache of the user browser and not necessary fetch (so the bug is still here).

File: tasks/build-versions-comparison.js (hunk 1 above)
Comment (NolwennP) [reply tasks/build-versions-comparison.js:18]: 👍

File: src/whats-new.handlebars [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+        });
+        _.each(domains,function(domain){
+            var checked = _.find(checkedDomains, function(checkedDomain) { return checkedDomain === domain; }) ? true : false;
+            document.getElementById(domain).checked = checked;
+        });
+    }
+
+    function countNews(){
+        var newsCount = $grid.data('isotope').filteredItems.length;
+        document.getElementById('news-count').innerHTML = newsCount ? 'There were <b>' + newsCount + '</b> new key features delivered in the PIM for the versions, edition and domains you selected. 💪' :  'Uh oh! There were <b>' + newsCount + '</b> new key feature delivered in the PIM for the versions, edition and domains you selected.<br/><b>Don\'t hesitate to change the above filters to see more new features!</b> 😉';
```
Comment (camillebourjade) [inline src/whats-new.handlebars:127]:
```suggestion
        document.getElementById('news-count').innerHTML = newsCount ? 'There were <b>' + newsCount + '</b> new key features delivered in the PIM for the versions, edition and domains you selected. 💪' :  'Uh oh! There were <b>' + newsCount + '</b> new key feature delivered in the PIM for the versions, edition and domains you selected.<br/><b>Don\'t hesitate to play around with the other filters to see more new features!</b> 😉';
```

**Review by camillebourjade — COMMENTED** [review]
English Fix!

**Pull request merged**